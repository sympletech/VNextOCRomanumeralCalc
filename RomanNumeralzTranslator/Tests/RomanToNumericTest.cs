using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace RomanNumeralzTranslator
{
    [UseReporter(typeof(DiffReporter))]

    [TestClass]
    public class RomanToNumericTest
    {
        
        [TestMethod]
        public void RomanToDecimalTest()
        {
            string[] romanDigits = new[]
            {
                "I", "II", "III", "IV", "V", "VI", "VII", "VIII","IX","X",
                "XI", "XII", "XIII", "XIV", "XV", "XVI", "XVII", "XVIII","XIX","XX",
                "XXX","XXXIX","", "XL", "XLIX", "L", "l", "XC", "C",
                "XLVIII"
                
            };
            var results = romanDigits.Select(x => new {key = x, value = RomanNumeralTranslator.Translate(x)});
            ApprovalTests.Approvals.VerifyAll(results, "");
        }

        [TestMethod]
        public void RomanToDecimalExceptionTest()
        {
            var ex = ExceptionUtilities.GetException(() => RomanNumeralTranslator.Translate("Z"));
            Approvals.Verify(ex);
        }

        [TestMethod]
        public void BadDigitsTest()
        {
            string[] badDigits = {"IIX","IIZ","MIM", "MOM", "IVI", "HIV","XXVX", "VV","LL","LLCOOLJ"};
            var ex =
                badDigits.Select(
                    x =>
                        new
                        {
                            Key = x,
                            Value = ExceptionUtilities.GetException(() => RomanNumeralTranslator.Translate(x))
                        });
            Approvals.VerifyAll(ex,"");
        }
    }
}