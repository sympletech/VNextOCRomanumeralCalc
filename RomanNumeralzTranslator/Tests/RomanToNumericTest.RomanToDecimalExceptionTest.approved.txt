System.ArgumentException: Z not a valid Roman numeral
   at RomanNumeralzTranslator.RomanNumeralTranslator.ThrowOnInvalidToken(String token) in ...\RomanNumeralTranslator.cs
   at RomanNumeralzTranslator.RomanNumeralTranslator.Translate(String roman) in ...\RomanNumeralTranslator.cs
   at RomanNumeralzTranslator.RomanToNumericTest.<RomanToDecimalExceptionTest>() in ...\RomanToNumericTest.cs
   at ApprovalUtilities.Utilities.ExceptionUtilities.GetException(Action action) in ...\ExceptionUtilities.cs